frontend/finance-tracker/src/components/ui/calendar.tsx

```
@@ -2,74 +1,524 @@
+} & React.HTMLAttributes<HTMLDivElement>) {
+    const { goToMonth, selected } = useDayPicker();
+
+    return (
+        <div className={cn("grid grid-cols-4 gap-y-2", className)} {...props}>
+            {Array.from(
+                { length: displayYears.to - displayYears.from + 1 },
+                (_, i) => {
+                    const isBefore =
+                        differenceInCalendarDays(
+                            new Date(displayYears.from + i, 11, 31),
+                            startMonth!
```

Biome (1.9.4)

[error] Forbidden non-null assertion.



(lint/style/noNonNullAssertion)